<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FieldDefinitionDialog</class>
 <widget class="QDialog" name="FieldDefinitionDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>New Field Definition</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="nameLabel">
       <property name="text">
        <string>Name*</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="nameLineEdit"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="descriptionLabel">
       <property name="text">
        <string>Description</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="descriptionLineEdit"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="frameTypeLabel">
       <property name="text">
        <string>Which contains*</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QRadioButton" name="commonFieldTypeRadioButton">
         <property name="text">
          <string>None</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="bitfieldsFieldTypeRadioButton">
         <property name="text">
          <string>Bitfields</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="subframesFieldTypeRadioButton">
         <property name="text">
          <string>Subframes</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="bytesLineEdit"/>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="bytesLabel">
       <property name="text">
        <string>Bytes*</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QStackedWidget" name="stackedWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="commonFieldTypePage">
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="decoderLabel">
           <property name="text">
            <string>Decoder</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="decoderLineEdit"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="defaultsLabel">
           <property name="text">
            <string>Defaults</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="defaultsLineEdit"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="subframeFieldTypePage">
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QLabel" name="subframesLabel">
         <property name="text">
          <string>Subframes To Be Detected*</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QTreeView" name="subframesView">
           <property name="editTriggers">
            <set>QAbstractItemView::NoEditTriggers</set>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <widget class="QPushButton" name="addSubframeButton">
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="deleteSubframeButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Delete</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="editSubframeButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Edit</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="moveUpSubframeButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Move Up</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="moveDownSubframeButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Move Down</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="bitfieldFieldTypePage">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="bitfieldDefinitionsLabel">
         <property name="text">
          <string>Bitfield Definitions*</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QTreeView" name="bitfieldDefinitionsView">
           <property name="editTriggers">
            <set>QAbstractItemView::NoEditTriggers</set>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QPushButton" name="addBitfieldButton">
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="deleteBitfieldButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Delete</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="editBitfieldButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Edit</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="moveUpBitfieldButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Move Up</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="moveDownBitfieldButton">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Move Down</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="errorLabel">
     <property name="text">
      <string/>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
